src/core/DomainDecomposition.hpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (C) 2010-2020 The ESPResSo project
+ * Copyright (C) 2002,2003,2004,2005,2006,2007,2008,2009,2010
+ *   Max-Planck-Institute for Polymer Research, Theory Group
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef ESPRESSO_DOMAIN_DECOMPOSITION_HPP
+#define ESPRESSO_DOMAIN_DECOMPOSITION_HPP
+
+#include "ParticleDecomposition.hpp"
+
+#include "BoxGeometry.hpp"
+#include "LocalBox.hpp"
+
+/** @brief Structure containing the information about the cell grid used for
+ * domain decomposition.
+ *
+ *  The domain of a node is split into a 3D cell grid with dimension
+ *  cell_grid. Together with one ghost cell
+ *  layer on each side the overall dimension of the ghost cell grid is
+ *  ghost_cell_grid. The domain
+ *  decomposition enables one the use of the linked cell algorithm
+ *  which is in turn used for setting up the Verlet list for the
+ *  system. You can see a 2D graphical representation of the linked
+ *  cell grid below.
+ *
+ *  \image html  linked_cells.gif "Linked cells structure"
+ *
+ *  2D representation of a linked cell grid: cell_grid =
+ *  {4,4}, ghost_cell_grid = {6,6}
+ *
+ * Each cell has 3^D neighbor cells (For cell 14 they are
+ * marked). Since we deal with pair forces, it is sufficient to
+ * calculate only half of the interactions (Newtons law: action =
+ * reaction). We have chosen the upper half e.g. all neighbor cells with
+ * a higher linear index (For cell 14 they are marked in light
```

Comment:
```suggestion
 * Each cell has 3^D neighbor cells. Since we deal with pair forces, it is
 * sufficient to calculate only half of the interactions (Newton's law:
 * action = reaction). We have chosen the upper half e.g. all neighbor
 * cells with a higher linear index (for cell 14 they are marked in light
```
remove typo and text duplication